les/peer.go

```diff
@@ -580,8 +600,10 @@ func (ps *peerSet) Unregister(id string) error {
 		for _, n := range peers {
 			n.unregisterPeer(p)
 		}
+		p.lock.Lock()
```

Comment:
Do not use peer.lock here, it does not protect anything that is accessed here. If there is a data race with sendQueue then the right solution is to move its creation (see line `p.sendQueue = newExecQueue(100)`) from peerSet.Register to newPeer so that it is never changed during the lifetime of the peer and it does not need locking at all. Maybe open a separate PR if it has nothing to do with ULC. It is generally a good practice to not include unrelated bugfixes in a PR because it is really hard to find the cause if something goes wrong.